### ISSUE TITLE
There should be cop to detect method call using class name in the same class.

### ISSUE THREAD

**Issue by akshaymohite** [rubocop-hq/ruby-style-guide#376]
There should be cop to detect method call with class in name from the class method of the same class. E.g.

``` Ruby
def TestClass
  def self.some_method_a
    # some code
    some_method_c
  end

  def self.some_method_b
    # some code
    TestClass.some_method_c
  end

  def self.some_method_c
    # some code
  end
end
```

Cop should detect that `TestClass.some_method_c` calling method `some_method_c` should be directly instead of `ClassName.method_name`.
As if the method `some_method_c` would have been private, this call `TestClass.some_method_c` will give error on the method call.
